WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -32,3 +84,91 @@ final class ProductDetailPreviewViewModel: ObservableObject {
         // TODO
     }
 }
+
+private extension ProductDetailPreviewViewModel {
+    func observeGeneratedProduct() {
+        generatedProductSubscription = $generatedProduct
+            .compactMap { $0 }
+            .sink { [weak self] product in
+                guard let self else { return }
+                self.updateProductDetails(with: product)
+            }
+    }
+
+    func updateProductDetails(with product: Product) {
+        productName = product.name
+        productDescription = product.fullDescription ?? product.shortDescription ?? self.productDescription
+        productType = product.virtual ? Localization.virtualProductType : Localization.physicalProductType
+
+        if let regularPrice = product.regularPrice, regularPrice.isNotEmpty {
+            let formattedRegularPrice = currencyFormatter.formatAmount(regularPrice, with: currency) ?? ""
+            productPrice = String.localizedStringWithFormat(Localization.regularPriceFormat, formattedRegularPrice)
+        }
+
+        productCategories = product.categoriesDescription()
+        productTags = product.tagsDescription()
+        updateShippingDetails(for: product)
+    }
+
+    func updateShippingDetails(for product: Product) {
```

Comment:
Nit: What do you think about extracting this formatting code into a formatter class and reusing it?

Reply:
If you mean reusing the logic with Product Form, I tried doing that but could not because the form uses a different data model - so even though the code looks similar, we cannot reuse it unless we change the model used for this screen too. I'll keep this implementation as-is.

Reply:
Sure, thanks for the explanation. Let us keep it as-is. 👍